apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -1,17 +1,37 @@
 defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
   alias Future.Code, as: Code
   alias Lexical.Ast
+  alias Lexical.Ast.Analysis
   alias Lexical.Document
   alias Lexical.Document.Location
   alias Lexical.Document.Position
+  alias Lexical.Formats
   alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
   alias Lexical.Text
 
   def definition(%Document{} = document, %Position{} = position) do
-    document
+    with {:ok, _, analysis} <- Document.Store.fetch(document.uri, :analysis),
+         {:ok, entity, _range} <- Entity.resolve(analysis, position) do
+      fetch_definition(entity, analysis, position)
+    end
+  end
+
+  defp fetch_definition({type, entity}, %Analysis{} = _analysis, %Position{} = _position)
+       when type in [:struct, :module] do
+    module = Formats.module(entity)
+
+    case Store.exact(module, type: type, subtype: :definition) do
+      [entry] -> to_location(entry)
+      _other -> :error
```

Comment:
I think, per our discussion, this is also fine:
```elixir
module
|> Store.exact(type: type, subtype: :definition)
|> Enum.map(&to_location/1)
```

Reply:
![CleanShot 2024-04-12 at 16 10 22@2x](https://github.com/lexical-lsp/lexical/assets/12830256/6333467d-fd6b-463b-9168-6073b4f07263)
Feel free to try the last commit: 57c82f462e7ff6d67abfdf19f4ecf3a3932c6f7e, if no more issues, then I'll merge it.